File: src/main/java/com/manning/simplysend/mappers/ItemMapper.java
package com.manning.simplysend.mappers;

import com.manning.simplysend.dto.ItemDTO;
import com.manning.simplysend.entities.Item;

public class ItemMapper {

    public static Item fromDTO(ItemDTO itemDTO) {
        Item item = new Item();
        item.setId(itemDTO.getId());
        item.setName(itemDTO.getName());
        item.setDescription(itemDTO.getDescription());
        item.setPrice(itemDTO.getPrice());
        item.setType(itemDTO.getType());
        return item;
    }

    public static ItemDTO toDTO(Item item) {
        ItemDTO itemDTO = new ItemDTO();
        itemDTO.setId(item.getId());
        itemDTO.setName(item.getName());
        itemDTO.setDescription(item.getDescription());
        itemDTO.setPrice(item.getPrice());
        itemDTO.setType(item.getType());
        return itemDTO;
    }
}

File: src/main/kotlin/com/manning/simplysend/mappers/ItemMapper.kt
package com.manning.simplysend.mappers

import com.manning.simplysend.dto.ItemDTO
import com.manning.simplysend.entities.Item

object ItemMapper : BaseMapper() {

    fun fromDTO(itemDTO: ItemDTO): Item {
        val item = Item()
        item.id = itemDTO.id
        item.name = itemDTO.name
        item.description = itemDTO.description
        item.price = itemDTO.price
        item.type = itemDTO.type
        return item
    }

    fun toDTO(item: Item): ItemDTO {
        val itemDTO = ItemDTO()
        itemDTO.id = item.id
        itemDTO.name = htmlEscape(item.name)
        itemDTO.description = htmlEscape(item.description)
        itemDTO.price = item.price
        itemDTO.type = item.type
        return itemDTO
    }
}